### PR TITLE
qt5: 5.9.1 -> 5.9.3

### PR DESCRIPTION
**Pull request by bkchr**
###### Motivation for this change
This is just a first blocker pull request. I'm still waiting on `nox-review` to finish. After I fixed all packages, I will split them into individual pull requests.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/misc/redis-desktop-manager/breakpad_ucontext.patch [hunk 1]
```diff
@@ -0,0 +1,223 @@
+Submodule 3rdparty/gbreakpad contains untracked content
```
Comment (Mic92) [inline pkgs/applications/misc/redis-desktop-manager/breakpad_ucontext.patch]: Did you write this patch on your own?

File: pkgs/tools/security/qesteidutil/breakpad_ucontext.patch [hunk 2]
```diff
@@ -0,0 +1,202 @@
+diff --git a/common/google-breakpad/src/client/linux/dump_writer_common/ucontext_reader.cc b/common/google-breakpad/src/client/linux/dump_writer_common/ucontext_reader.cc
```
Comment (Mic92) [inline pkgs/tools/security/qesteidutil/breakpad_ucontext.patch]: and this one?

**Comment by Mic92** [comment]
If possible please use `fetchpatch` to download fixes from upstream rather then including it into nixpkgs.

File: pkgs/applications/misc/redis-desktop-manager/breakpad_ucontext.patch (hunk 1 above)
Comment (bkchr) [reply pkgs/applications/misc/redis-desktop-manager/breakpad_ucontext.patch]: Backport of an upstream patch. Do I need to insert Copyright?

File: pkgs/tools/security/qesteidutil/breakpad_ucontext.patch (hunk 2 above)
Comment (bkchr) [reply pkgs/tools/security/qesteidutil/breakpad_ucontext.patch]: Same as above.

**Comment by bkchr** [comment]
@Mic92 I would like to do that, but can I specify to apply a patch to a sub directory?

File: pkgs/applications/misc/redis-desktop-manager/breakpad_ucontext.patch (hunk 1 above)
Comment (Mic92) [reply pkgs/applications/misc/redis-desktop-manager/breakpad_ucontext.patch]: No, this question was more related to https://github.com/NixOS/nixpkgs/pull/32100#issuecomment-347130991

**Comment by orivej** [comment]
> can I specify to apply a patch to a sub directory

One option is to add missing parent directories with the `addPrefixes` argument of `fetchpatch`.

**Comment by bkchr** [comment]
Okay, now everything is ready :)

**Comment by LnL7** [comment]
Testing on darwin...

**Review by LnL7 — COMMENTED** [review]
```
/nix/store/iycdv3n44xrnd3y0mi2i6h74g13595ba-apple-framework-AppKit/Library/Frameworks/AppKit.framework/Headers/NSWindow.h:307:41: note: property declared here
@property (assign) id<NSWindowDelegate> delegate;
                                        ^
In file included from qcocoaclipboard.mm:40:
In file included from ./qcocoaclipboard.h:47:
In file included from ./qmacclipboard.h:46:
In file included from /nix/store/iycdv3n44xrnd3y0mi2i6h74g13595ba-apple-framework-AppKit/Library/Frameworks/AppKit.framework/Headers/AppKit.h:88:
In file included from /nix/store/iycdv3n44xrnd3y0mi2i6h74g13595ba-apple-framework-AppKit/Library/Frameworks/AppKit.framework/Headers/NSPopUpButton.h:9:
/nix/store/iycdv3n44xrnd3y0mi2i6h74g13595ba-apple-framework-AppKit/Library/Frameworks/AppKit.framework/Headers/NSMenuItemCell.h:65:32: warning: attribute 'readonly' of property 'tag' restricts attribute 'readwrite' of property inherited from 'NSActionCell' [-Wproperty-attribute-mismatch]
@property (readonly) NSInteger tag;
                               ^
qcocoawindow.mm:323:5: error: type arguments cannot be applied to non-parameterized class 'NSEnumerator'
    NSEnumerator<NSWindow*> *windowEnumerator = nullptr;
    ^           ~~~~~~~~~~~
...
make[4]: Leaving directory '/private/tmp/nix-build-qtbase-5.9.3.drv-0/qtbase-opensource-src-5.9.3/src/plugins/platforms/cocoa'
make[3]: *** [Makefile:97: sub-cocoa-make_first] Error 2
make[3]: Leaving directory '/private/tmp/nix-build-qtbase-5.9.3.drv-0/qtbase-opensource-src-5.9.3/src/plugins/platforms'
make[2]: *** [Makefile:102: sub-platforms-make_first] Error 2
make[2]: Leaving directory '/private/tmp/nix-build-qtbase-5.9.3.drv-0/qtbase-opensource-src-5.9.3/src/plugins'
make[1]: *** [Makefile:695: sub-plugins-make_first] Error 2
make[1]: Leaving directory '/private/tmp/nix-build-qtbase-5.9.3.drv-0/qtbase-opensource-src-5.9.3/src'
make: *** [Makefile:49: sub-src-make_first] Error 2
builder for '/nix/store/9197vkq40xbnqkkhcbljv9ysq50xa0r0-qtbase-5.9.3.drv' failed with exit code 2
```

**Comment by bkchr** [comment]
@LnL7 I don't have any mac, can you try to fix the errors?

File: pkgs/development/libraries/qt-5/5.9/qtbase.patch [hunk 3]
```diff
@@ -1001,30 +1028,8 @@ index c92d8fc3f8..6008063bcf 100644
...
-diff --git a/src/plugins/bearer/corewlan/qcorewlanengine.mm b/src/plugins/bearer/corewlan/qcorewlanengine.mm
-index 341d3bccf2..3368234c26 100644
---- a/src/plugins/bearer/corewlan/qcorewlanengine.mm
-+++ b/src/plugins/bearer/corewlan/qcorewlanengine.mm
-@@ -287,7 +287,7 @@ void QScanThread::getUserConfigurations()
-     QMacAutoReleasePool pool;
-     userProfiles.clear();
- 
--    NSArray<NSString *> *wifiInterfaces = [CWWiFiClient interfaceNames];
-+    NSArray *wifiInterfaces = [CWWiFiClient interfaceNames];
```
Comment (LnL7) [inline pkgs/development/libraries/qt-5/5.9/qtbase.patch]: Why did you remove these hunks? That probably explains the darwin failure.

Comment (LnL7) [reply pkgs/development/libraries/qt-5/5.9/qtbase.patch]: Could you add those back, I will try to fix the other issues if it's still broken.

Comment (bkchr) [reply pkgs/development/libraries/qt-5/5.9/qtbase.patch]: Done

**Review by LnL7 — APPROVED** [review]

**Comment by orivej** [comment]
This broke creating directories in KDE: #32253

**Comment by bkchr** [comment]
:(

**Comment by orivej** [comment]
Reverted to Qt 5.9.1 in #32100. Nevertheless, thank you for preparing the upgrade, we will restore it later!

**Pull request merged**